src/main/java/io/vertx/core/eventbus/impl/HandlerRegistration.java

```diff
@@ -377,4 +343,82 @@ public Object getMetric() {
     return metric;
   }
 
+  protected class InboundDeliveryContext implements DeliveryContext<T> {
+
+    private final MessageImpl<?, T> message;
+    private final Iterator<Handler<DeliveryContext>> iter;
+    private final Handler<Message<T>> handler;
+    private final ContextInternal context;
+
+    private InboundDeliveryContext(MessageImpl<?, T> message, Handler<Message<T>> handler, ContextInternal context) {
+      this.message = message;
+      this.handler = handler;
+      this.iter = eventBus.receiveInterceptors();
+      this.context = message.src ? context : context.duplicate();
+    }
+
+    @Override
+    public Message<T> message() {
+      return message;
+    }
+
+    @Override
+    public void next() {
+      if (iter.hasNext()) {
+        try {
+          Handler<DeliveryContext> handler = iter.next();
+          if (handler != null) {
+            handler.handle(this);
+          } else {
+            next();
+          }
+        } catch (Throwable t) {
+          log.error("Failure in interceptor", t);
+        }
+      } else {
+        boolean local = true;
+        if (message instanceof ClusteredMessage) {
+          // A bit hacky
+          ClusteredMessage cmsg = (ClusteredMessage)message;
+          if (cmsg.isFromWire()) {
+            local = false;
+          }
+        }
+        try {
+          if (metrics != null) {
+            metrics.beginHandleMessage(metric, local);
+          }
+          VertxTracer tracer = handlerContext.tracer();
+          if (tracer != null && !src) {
+            List<Map.Entry<String, String>> tags = Collections.singletonList(new AbstractMap.SimpleEntry<>("peer.service", message.address));
```

Comment:
imho (and in zipkin) the remote address is important enough to be a type. It is easier to change a type to a named tag vs the other way around. Also the name "service" is usually about the friendly name, not the IP.

https://github.com/openzipkin/brave/blob/master/brave/src/main/java/brave/Span.java#L168

Reply:
further, it would be interesting to know if this type of property (whether escalated to its own callback or not) is always there or only on remotely originated requests.

Reply:
"peer.address" may be generic enough for this message address. Here's a reference to semantic conventions used in open tracing - https://github.com/opentracing/specification/blob/master/semantic_conventions.md#span-tags-table

Reply:
in this case, the message.address is the name under which the event-bus consumer was registered, i.e `eventBus.consumer("foo", handler);` then `peer.address` is `foo`, how would it be translated in Zipkin ?